distribution/pull_v2.go

```diff
@@ -393,7 +393,7 @@ func (p *v2Puller) pullV2Tag(ctx context.Context, ref reference.Named) (tagUpdat
 	oldTagImageID, err := p.config.ReferenceStore.Get(ref)
 	if err == nil {
 		if oldTagImageID == imageID {
-			return false, nil
+			return false, addDigestReference(p.config.ReferenceStore, ref, manifestDigest, imageID)
```

Comment:
I think the worst thing that can happen here is that user can get a digest after a pull that is invalid, meaning that it doesn't reflect the image contents and pulling by this digest will fail. If someone now considers this digest as secure, if they use it to pull something again in a fresh machine they would bypass the first level of verification(compressed data) but still get an error on unpacking. Not sure if this is important enough to fix.